pyfr/solvers/navstokes/system.py

```diff
@@ -15,3 +15,66 @@ class NavierStokesSystem(BaseAdvectionDiffusionSystem):
     mpiinterscls = NavierStokesMPIInters
     bbcinterscls = NavierStokesBaseBCInters
 
+    def rhs(self, t, uinbank, foutbank):
+        runall = self.backend.runall
+        q1, q2 = self._queues
+        kernels = self._kernels
+
+        self.eles_scal_upts_inb.active = uinbank
+        self.eles_scal_upts_outb.active = foutbank
+
+        q1 << kernels['eles', 'disu']()
+        q1 << kernels['mpiint', 'scal_fpts_pack']()
+        runall([q1])
+
+        q1 << kernels['iint', 'con_u']()
+        q1 << kernels['bcint', 'con_u']()
+        q1 << kernels['eles', 'tgradpcoru_upts']()
+        if ('eles', 'avis') in kernels:
+            q1 << kernels['eles', 'avis']()
+            q1 << kernels['mpiint', 'avis_fpts_pack']()
+
+        q2 << kernels['mpiint', 'scal_fpts_send']()
+        q2 << kernels['mpiint', 'scal_fpts_recv']()
+        q2 << kernels['mpiint', 'scal_fpts_unpack']()
+
+        runall([q1, q2])
+
+        q1 << kernels['mpiint', 'con_u']()
+        q1 << kernels['eles', 'tgradcoru_upts']()
+        q1 << kernels['eles', 'gradcoru_upts']()
+        q1 << kernels['eles', 'gradcoru_fpts']()
+        q1 << kernels['mpiint', 'vect_fpts_pack']()
+        if ('eles', 'avis') in kernels:
+            q2 << kernels['mpiint', 'avis_fpts_send']()
+            q2 << kernels['mpiint', 'avis_fpts_recv']()
+            q2 << kernels['mpiint', 'avis_fpts_unpack']()
+
+            runall([q1, q2])
```

Comment:
Should be able to do this in all cases.  An empty queue will not cause any issues.